adp/core/state.py

```
@@ -0,0 +1,220 @@
+    
+    # Define transition rules based on logical necessity
+    TRANSITION_RULES = [
+        TransitionRule(StateType.SHARED, StateType.SHARED, True, 
+                      "Shared remains shared"),
+        TransitionRule(StateType.SHARED, StateType.SEPARATE, False, 
+                      "Shared cannot become separate - fundamental axiom"),
+        TransitionRule(StateType.SEPARATE, StateType.SEPARATE, True, 
+                      "Separate remains separate"),
+        TransitionRule(StateType.SEPARATE, StateType.SHARED, True, 
+                      "Separate can be revealed to become shared"),
+    ]
```

Ruff (0.13.1)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)